### PR TITLE
Added models for the classes "Items" and "Users"

### PR DESCRIPTION
**Pull request by elaverty0103**

### CODE REVIEW

File: src/models/user.repository.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+  if (!userExists) {
+    throw new Error("User does not exist");
+  }
+  // Create item
+  const newItem = await Item.create({
+    ...item,
+    userId: userExists.username,
+  });
+  return newItem;
+};
```
Comment (santiagomed) [inline src/models/user.repository.ts:90]: I know I included these two functions in this repository for the backend workshop, but part of me wonders if it is best to keep item queries only in the item repository. We can keep this here for now, but it's something to think about later on.

File: src/models/user.repository.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+  }
+  // Create user
+  const newUser = await User.create(user);
+  return newUser;
+};
+
+/**
+ * @desc Check if user exists and return user
+ * @param {User} user object
+ * @returns {Promise<User | null>} Username or false if user does not exist or password is incorrect
```
Comment (santiagomed) [inline src/models/user.repository.ts]:
```suggestion
 * @returns {Promise<User | null>} Username or null if user does not exist or password is incorrect
```

File: src/models/user.repository.ts [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+ */
+export const loginUser = async (user: User): Promise<User | null> => {
```
Comment (santiagomed) [inline src/models/user.repository.ts]:
```suggestion
export const userExists = async (user: User): Promise<User | null> => {
```

Should probably change this function name to `userExists` or something similar; we are not actually logging in a user in this function.

File: src/models/user.repository.ts [hunk 4]
```diff
@@ -0,0 +1,90 @@
+import { User } from "./models";
+import { Item } from "./models";
+
+/**
+ * @desc Create a new user
+ * @param {User} user
+ * @returns {Promise<User | boolean>} Created user or false if user already exists
+ */
+export const createUser = async (user: User): Promise<User | boolean> => {
```
Comment (santiagomed) [inline src/models/user.repository.ts:9]: Might be worth it to look into an alternate implementation that just returns one specific data type.

**Review by santiagomed — COMMENTED** [review]
These changes are not pressing, but feel free to commit my suggestions.

**Review by santiagomed — APPROVED** [review]
Besides some of the comments I left, this PR is ready for merge!

**Review by jennweir — APPROVED** [review]
Looks good but I agree we should keep the comments in mind.

**Pull request merged**